### PR TITLE
Fix issues in TorpadoStore

### PR DESCRIPTION
**Pull request by botondsulyok**
Fix 3 issues in TorpadoStore with the help of Sonar.

### CODE REVIEW

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 1]
```diff
@@ -14,6 +14,8 @@
 
   private int torpedoCount = 0;
 
+  private Random generator = new Random();
```
Comment (labancztamas) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:17]: Good idea!

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 2]
```diff
@@ -30,18 +32,17 @@ public TorpedoStore(int numberOfTorpedos){
 
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
```
Comment (labancztamas) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:36]: Nice one!

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 3]
```diff
@@ -30,18 +32,17 @@ public TorpedoStore(int numberOfTorpedos){
...
     boolean success = false;
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
     double r = generator.nextDouble();
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
+      this.torpedoCount -= numberOfTorpedos;
```
Comment (labancztamas) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:46]: You are right!

**Review by labancztamas — CHANGES_REQUESTED** [review]
Perfect!

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java (hunk 2 above)
Comment (botondsulyok) [reply src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:36]: Thank you.

**Review by labancztamas — APPROVED** [review]

**Pull request merged**